src/Cli/dotnet/commands/dotnet-test/IPC/Serializers/BaseSerializer.cs

```diff
@@ -218,16 +218,28 @@ protected static bool ReadBool(Stream stream)
 
     protected static void WriteField(Stream stream, ushort id, string? value)
     {
-        if (value is null)
+        if (IsNull(value))
         {
             return;
         }
 
         WriteShort(stream, id);
-        WriteStringSize(stream, value);
-        WriteString(stream, value);
+        WriteStringSize(stream, value!);
+        WriteString(stream, value!);
     }
 
+    protected static void WriteField(Stream stream, string? value)
+    {
+        if (IsNull(value))
+        {
+            return;
+        }
+
+        WriteString(stream, value!);
+    }
+
+    protected static bool IsNull(string? value) => value is null;
```

Comment:
I would inline this check so we don't need the `value!` after the check. Let's remove this one liner helper.